.github/workflows/test.yml

```
@@ -24,3 +24,10 @@ jobs:
 
             - name: Test with tox
               run: tox
+
+            - name: Generate apidoc
+              run: tox -e docs
+
+            - name: Check if repo apidoc is up to date
+              run: |
+                git diff-index --quiet HEAD --
```

```suggestion
                git diff-index HEAD
```
I think this is enough, also no need to suppress output

---

what's the trailing `--` for?

---

idk i just pasted it lol, ill remove it though